modules/renter/hostdb/persist.go

```diff
@@ -75,3 +71,27 @@ func (hdb *HostDB) load() error {
 	hdb.lastChange = data.LastChange
 	return nil
 }
+
+// threadedSaveLoop saves the hostdb to disk every 2 minutes, also saving when
+// given the shutdown signal.
+func (hdb *HostDB) threadedSaveLoop(shutChan chan struct{}) {
```

Comment:
IMO if you want to express "always save right before shutdown," the clearest way to do it is in `AfterStop` (or in the `Close` function itself :P)

This seems like a good change to roll out to other modules. Losing 2 minutes of data isn't a big deal as long as it can be recovered by resubscribing. There isn't much data that we absolutely have to `saveSync` as soon as it arrives. Just contracts and some wallet stuff I imagine.